plutus-ledger-api/test/Spec/Eval.hs

```diff
@@ -73,8 +84,48 @@ testUnlifting = testCase "check unlifting behaviour changes in Vasil" $ do
     evalAPI alonzoPV illPartialBuiltin @?= True
     evalAPI vasilPV illPartialBuiltin @?= True
 
+costParams :: [Integer]
+costParams = Map.elems (fromJust defaultCostModelParams)
+
+lengthParamNamesV :: PlutusLedgerLanguage -> Int
+lengthParamNamesV PlutusV1 = length $ enumerate @V1.ParamName
+lengthParamNamesV PlutusV2 = length $ enumerate @V2.ParamName
+lengthParamNamesV PlutusV3 = length $ enumerate @V3.ParamName
+
+mkEvaluationContextV :: PlutusLedgerLanguage -> IO EvaluationContext
+mkEvaluationContextV lv =
+    either (assertFailure . display) (pure . fst) . runWriterT $
+        take (lengthParamNamesV lv) costParams & case lv of
+            PlutusV1 -> V1.mkEvaluationContext
+            PlutusV2 -> V2.mkEvaluationContext
+            PlutusV3 -> V3.mkEvaluationContext
+
+-- | Ensure that 'toMachineParameters' never throws for all language and protocol versions.
+evaluationContextCacheIsComplete :: TestTree
+evaluationContextCacheIsComplete =
+    testGroup "EvaluationContext has machine parameters for all protocol versions" $
+        enumerate <&> \lv -> testCase (show lv) $ do
```

Comment:
could make it a property over language versions and protocol versions? probably not worth it

Reply:
Why? It's computationally trivial to enumerate them all. Once we have 10k lvs * pvs, then maybe we should think about optimizing it, but right now I'm definitely not gonna introduce any unnecessary risk by reducing test coverage.